### ISSUE TITLE
Make max retry attempts and sleep interval configurable

### ISSUE THREAD

**Issue by andreykaipov** [gruntwork-io/terragrunt#1630]
I saw from [your docs](https://terragrunt.gruntwork.io/docs/features/auto-retry/) this is something you'd be open to:

> Future upgrades to terragrunt may include the ability to configure max retries and retry intervals in the terragrunt config (PRs welcome!).

My corp network is absolute dogshit so even three retries is not enough to save our infrastructure, so I'd like to help!

I'll submit a PR later today but figured I'd open up this issue too for tracking purposes.

**Comment by andreykaipov** [comment]
I was thinking of just adding two more top-level fields, e.g.:

```hcl
max_retry_attempts = 5
retry_sleep_interval = 10
retryable_errors = [
  "blah",
]
```

That feels like the most straight-forward approach, but lemme know if I should consider something else!

**Comment by brikis98** [comment]
Thanks!

**Comment by brikis98** [comment]
Fixed in #1632 and released in https://github.com/gruntwork-io/terragrunt/releases/tag/v0.28.19.
